Sources/SwiftSyntax/SyntaxProtocol.swift

```diff
@@ -238,6 +238,18 @@ extension SyntaxProtocol {
   public var previousToken: TokenSyntax? {
     return self.previousToken(viewMode: .sourceAccurate)
   }
+
+  /// Returns this node or the first ancestor that satisfies `condition`.
+  public func ancestorOrSelf<T>(mapping map: (Syntax) -> T?) -> T? {
```

Comment:
If we’re making this public API, could you add an entry to the [release notes](https://github.com/swiftlang/swift-syntax/blob/main/Release%20Notes/601.md)?

Reply:
I added it to the release notes. I hope the description is sufficient.

Reply:
Release notes look good. Thank you!